phd-tests/tests/src/boot_order/efi_utils.rs

```diff
@@ -363,20 +363,32 @@ pub(crate) async fn write_efivar(
 
     let mut new_value = attrs;
     new_value.extend_from_slice(data);
+    let data_len = new_value.len();
 
     // The command to write this data back out will be, roughtly:
     // ```
-    // printf "\xAA\xAA\xAA\xAA\xDD\xDD\xDD\xDD" > /sys/firmware/efi/efivars/...
+    // printf "\xAA\xAA\xAA\xAA\xDD\xDD\xDD\xDD" | \
+    //   dd obs={inlen} of=/sys/firmware/efi/efivars/... status=none
     // ```
     // where AAAAAAAA are the attribute bytes and DDDDDDDD are caller-provided
     // data.
+    //
+    // notably do not printf directly to /sys/firmware/efi/efivars/*!! printf
+    // may flush output early if the data to write contains a `\n` (observed at
+    // least on Debian 11), and such a partial write to efivars may be rejected
+    // as invalid UEFI variable data.
```

Comment:
Is this something that's induced by our current long-command handling? I'm trying to decide how much I should try to suppress the urge to overhaul PHD's serial console and `run_shell_command` abstractions (in my copious free time, you understand).

Reply:
fully unrelated! i'm really curious why `printf` is now sufficient, or what made it insufficient in the past, but it's definitely a shell behavior that's flushing a partial line. for a very bad time, you can see shadows of this in search results about people having issues with `printf blah > /<efivarfs>/whatever` returning `invalid argument` sporadically, and lots of very strange suggestions on how to "fix the error".. i got very "how did this ever work" about it once i figured out the bug.

the long-command handling just sneaks a long shell line into the guest without prompting a `\n` back, so it'd still happen even if that were fixed, and i don't think we'd use much different commands in that world either. but the `run_long_command` stuff is very much a gross hack i'm urged to fix also.